cmd/syncthing/decrypt/decrypt.go

```diff
@@ -285,3 +293,27 @@ func loadEncryptedFileInfo(fd fs.File) (*protocol.FileInfo, error) {
 
 	return &encFi, nil
 }
+
+func (c *CLI) reconstructName(name, path string) string {
```

Comment:
I'd like to see some tests on this, it's not entirely obvious to me this is correct and it would be easy to exercise in a table driven test